bindata/network-diagnostics/network-check-source.yaml

```diff
@@ -0,0 +1,92 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: network-check-source
+  namespace: openshift-network-diagnostics
+  annotations:
+    kubernetes.io/description: |
+      This deployment deploys the network-check-source pod that performs
+      pod network connectivity checks
+    release.openshift.io/version: "{{.ReleaseVersion}}"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: network-check-source
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: network-check-source
+    spec:
+      serviceAccountName: network-diagnostics
+      containers:
+      - name: check-endpoints
+        image: "{{.NetworkCheckSourceImage}}"
+        imagePullPolicy: IfNotPresent
+        terminationMessagePolicy: FallbackToLogsOnError
+        command:
+          - cluster-network-check-endpoints
+        args:
+          - --listen
+          - 0.0.0.0:17698
+          - --namespace
+          - $(POD_NAMESPACE)
+        env:
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+        ports:
+          - name: check-endpoints
+            containerPort: 17698
+            protocol: TCP
+        resources:
+          requests:
+            memory: 50Mi
+            cpu: 10m
+
+---
+apiVersion: v1
```

Comment:
If this is the traffic source, why does it need a Service?
If the service is just for metrics, then it should be a headless service.

Reply:
Ah good point, it's indeed for Prometheus, I'll change.

Reply:
done